### PR TITLE
Npm

### PR DESCRIPTION
**Pull request by brettz9**
Builds on #4 and #5.

- Linting (ESLint): Update per latest eslint recommended
- npm: Add recommended properties (`contributors`, `dependencies`)
- npm: Add `engines` pointing tentatively to >=6.0.0
- npm: Update devDeps (resolves vulnerabilities)

I arbitrary chose 6.0.0 for `engines` as it is around the version which you already support (e.g., your use of `reduce` wouldn't allow the earliest Node versions I think, though https://node.green doesn't cover that), yet it is high enough to possibly also start using some other nice features. Let me know if you want to bump that even higher, e.g., to Node 8 or even 10 (as 8 has just reached End-of-Life, some projects are dropping support). Setting up Travis/CI would help to make sure we are indeed passing tests there if you felt inclined to set that up.

### CODE REVIEW

File: lib/rules/no-unsupported-keywords.js [hunk 1]
```diff
@@ -16,12 +16,12 @@ function checkExpression(context, expression, allKeywords) {
   var calleeText = source.getText(expression);
   var assertionText = calleeText.substr(expectText.length + 1);
 
-  var keywords = assertionText.split(/[\.\s]/).filter(function (k) {
+  var keywords = assertionText.split(/[.\s]/).filter(function (k) {
```
Comment (gavinaiken) [inline lib/rules/no-unsupported-keywords.js:19]: I know `.` doesn't actually need to be escaped in a `[]` set but I prefer to do it anyway for clarity and to avoid mistakes since outside of the `[]` context it would need to be escaped if the intention is to only match a `.`. So would prefer this line unchanged.

File: lib/rules/no-unsupported-keywords.js [hunk 2]
```diff
@@ -16,12 +16,12 @@ function checkExpression(context, expression, allKeywords) {
...
     return !!k;
   });
   for (var i = 0; i < keywords.length; i++) {
     var keyword = keywords[i];
-    if (!allKeywords.hasOwnProperty(keyword)) {
+    if (!{}.hasOwnProperty.call(allKeywords, keyword)) {
```
Comment (gavinaiken) [inline lib/rules/no-unsupported-keywords.js:24]: Why this change? Same functionality, the original is slightly more readable IMO. Presume your version has an advantage though?

File: package.json [hunk 3]
```diff
@@ -21,9 +24,13 @@
   },
   "homepage": "https://github.com/gavinaiken/eslint-plugin-chai-expect-keywords",
   "bugs": "https://github.com/gavinaiken/eslint-plugin-chai-expect-keywords/issues",
+  "engines": {
+    "node": ">=6.0.0"
```
Comment (gavinaiken) [inline package.json:28]: I'm using 10 or 12 in all my projects now so would be fine for me to go even higher, although probably keeping the support as broad as possible would benefit the general community. Probably bumping this to 8 would be a good place to land, although 6 is fine with me too.

**Review by gavinaiken — CHANGES_REQUESTED** [review]

**Comment by brettz9** [comment]
I've rebased this and the other PRs. (And it looks like #10 can be closed if it is not closed automatically after this PR.)

**Comment by brettz9** [comment]
Oops, hadn't seen your comments here, just a minute.

File: lib/rules/no-unsupported-keywords.js (hunk 1 above)
Comment (brettz9) [reply lib/rules/no-unsupported-keywords.js:19]: This is part of the new eslint recommended rules: https://eslint.org/docs/rules/no-useless-escape . Do you want me to disable this rule in your config then (or inline)?

**Comment by gavinaiken** [comment]
Sorry, just working my way through your PRs one by one

File: lib/rules/no-unsupported-keywords.js (hunk 2 above)
Comment (brettz9) [reply lib/rules/no-unsupported-keywords.js:24]: This is part of the rule added to eslint-recommended, at https://eslint.org/docs/rules/no-prototype-builtins . See that page for the rationales.

File: lib/rules/no-unsupported-keywords.js (hunk 1 above)
Comment (gavinaiken) [reply lib/rules/no-unsupported-keywords.js:19]: Ha! Didn't spot that. Nah if that is what eslint is recommending, let's go with it. No point arguing with a linter... ;)

File: package.json (hunk 3 above)
Comment (brettz9) [reply package.json:28]: Ok, bumping to 8 is nice for the features that could be used. Could do a little ES6 refactoring (some would work with just 6, but 8 is better of course).

File: lib/rules/no-unsupported-keywords.js (hunk 1 above)
Comment (brettz9) [reply lib/rules/no-unsupported-keywords.js:19]: lol :)

File: lib/rules/no-unsupported-keywords.js (hunk 2 above)
Comment (gavinaiken) [reply lib/rules/no-unsupported-keywords.js:24]: I see. Fine to roll with that, in that case.

Comment (brettz9) [reply lib/rules/no-unsupported-keywords.js:24]: But actually, it may become a non-issue later if I bump to Node 8 and then refactor this code to just use a for-of loop :)  That ensures it only iterates the "own" properties, so need for the extra check anyways.

File: package.json (hunk 3 above)
Comment (gavinaiken) [reply package.json:28]: I've merged this one now, could bump up further and use ES6 features later

Comment (brettz9) [reply package.json:28]: Or if you don't mind, if I could bump to 8.3, as that started support for object rest properties which can be very handy in such refactoring.
